project3/participant/tasks/multicast_receiver_task.cpp

```diff
@@ -17,26 +17,28 @@ thread_pool::Task::Status MulticastReceiver::SetUp() {
 }
 
 thread_pool::Task::Status MulticastReceiver::RunAtomic() {
-  parser_.ResetParser();
-  while (!parser_.HasCompleteMessage()) {
-    incoming_msg_buf_.resize(kBufferSize);
-    const auto bytes_read = participant_util::ReceiveForever(
-        messenger_fd_, incoming_msg_buf_.data(), kBufferSize, 0);
-    if (bytes_read <= 0) {
-      break;
-    }
-    incoming_msg_buf_.resize(bytes_read);
-    parser_.AddNewData(incoming_msg_buf_);
-  }
-  // empty buffer
-  incoming_msg_buf_.clear();
+  // If parser_ has a completed message ready
   if (parser_.HasCompleteMessage()) {
     pub_sub_messages::ForwardMulticast msg;
     parser_.GetMessage(&msg);
     std::string to_out =
         "[" + std::to_string(msg.origin_id()) + "] " + msg.message();
     LOG_S(0) << to_out;
     console_task_->SendConsole(to_out);
+    parser_.ResetParser();
```

Comment:
Technically, `GetMessage` does this internally, so this line should be superfluous.